### PR TITLE
Added users/self route. 

### LINKED ISSUE #37: Create a /users/self endpoint
When a logged in user hits this endpoint, they get details about themselves

This is especially useful when the user's new account gets created, but they have not selected a user name yet

### PR DESCRIPTION
**Pull request by Rucha1499**
Added the users/self route.  
Fixes #37 
This will return user object when user hits this endpoint.

### CODE REVIEW

File: controllers/usersController.js [hunk 1]
```diff
@@ -47,6 +47,17 @@ const getUser = async (req, res) => {
...
 
+/**
+ * Fetches the data about logged in user
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+
+const getSelfDetails = (req, res) => {
+  res.send(req.userData.user)
```
Comment (ankurnarkhede) [inline controllers/usersController.js:58]: - Add safe checks to check if the userData actually exists.

Comment (ankurnarkhede) [reply controllers/usersController.js:58]: return res.json(...)
Ideally the CI should have failed because of no return statement here. 🤔

Comment (ankurnarkhede) [reply controllers/usersController.js:58]: Add Tests.

**Review by ankurnarkhede — CHANGES_REQUESTED** [review]

**Comment by whyDontI** [comment]
@Rucha1499 I think you are working over swaraj's commits.
His PR is merged now, it'll be great if you can take a pull from Develop and only push your commits

File: controllers/usersController.js (hunk 1 above)
Comment (Rucha1499) [reply controllers/usersController.js]: I have added  the safe checks and return statement. I will add the tests soon.

File: controllers/usersController.js [hunk 2]
```diff
@@ -47,6 +47,25 @@ const getUser = async (req, res) => {
...
+
+const getSelfDetails = (req, res) => {
+  try {
+    if (req.userData) {
+      return res.send(req.userData)
+    }
+    return res.boom.notFound('User doesn\'t exist')
+  } catch (error) {
+    logger.error(`Error while fetching user: ${error}`)
+    return res.boom.serverUnavailable('Something went wrong please contact admin')
```
Comment (ankurnarkhede) [inline controllers/usersController.js:65]: `return res.boom.badImplementation`
Return 500 in this case.

File: routes/users.js [hunk 3]
```diff
@@ -7,6 +7,7 @@ const userValidator = require('../middlewares/validators/user')
 router.post('/', authenticate, userValidator.createUser, usersController.addNewUser)
 router.patch('/:id', authenticate, userValidator.updateUser, usersController.updateUser)
 router.get('/', authenticate, usersController.getUsers)
+router.get('/self', authenticate, usersController.getSelfDetails)
```
Comment (ankurnarkhede) [inline routes/users.js:10]: @Rucha1499 , can you add the API documentation annotations over this route?
Take reference from healthCheck.js

**Review by ankurnarkhede — CHANGES_REQUESTED** [review]

File: routes/users.js (hunk 3 above)
Comment (Rucha1499) [reply routes/users.js:46]: Done the changes regarding the returning of error. Added the API documentation annotations over the route users/self.

**Review by ankurnarkhede — APPROVED** [review]

**Comment by ankurnarkhede** [comment]
LGTM

**Pull request merged**